aaronlim02 iP

aaronlim02 opened the pull request:
# NotChatGPT
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

NotChatGPT is a task management application that allows users to create, manage, and track their daily tasks efficiently.

## Features:
- [x] Add/delete tasks
- [x] Mark Tasks as Complete
- [x] Due Dates
- [ ] Task prioritisation (Coming soon)

Why should you use it?
- Text-based
- Easy to learn
- ~~FAST~~ *SUPER FAST* to use :rocket:

All you need to do is:
1. Clone it from [GitHub Link](https://github.com/aaronlim02/ip.git).
2. Run it using Gradle.
3. Add your tasks.
4. Let it manage your tasks for you :wink:

And it is **FREE!**

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

aaronlim02 committed:
Completed Level 0. Rename, Greet, Exit

aaronlim02 committed:
completed Level 1. Echo

aaronlim02 committed:
completed Level 2. Add, List

aaronlim02 committed:
completed Level 3. Mark as Done

aaronlim02 committed:
Complete Level 4. ToDos, Events, Deadlines

aaronlim02 committed:
Complete A-TextUiTesting: Automated Text UI Testing

aaronlim02 committed:
Complete Level 5. Handle Errors

aaronlim02 committed:
Complete Level-6: Delete

aaronlim02 committed:
Create save functionality

aaronlim02 committed:
Fix file format

aaronlim02 committed:
Merge branch 'branch-Level-7'

aaronlim02 committed:
Fix load the data from the hard disk when the chatbot starts up

aaronlim02 committed:
Complete Level 8. Dates and Times

aaronlim02 committed:
Print date in a different format

aaronlim02 committed:
Test sample tasks

aaronlim02 committed:
Merge branch 'branch-Level-8'

aaronlim02 committed:
Add Increments: A-MoreOOP

aaronlim02 committed:
Merge branch 'branch-A-MoreOOP'

aaronlim02 committed:
Organize into Packages

aaronlim02 committed:
Merge branch 'add-gradle-support'

aaronlim02 committed:
Set package for class and Use Gradle

aaronlim02 committed:
Be able to run JUnit tests using Gradle

aaronlim02 committed:
Add JUnit Tests

aaronlim02 committed:
Add JUnit Tests

aaronlim02 committed:
Fix gradle to create a JAR File

aaronlim02 committed:
Add JavaDoc comments

aaronlim02 committed:
Follow the Coding Standard

aaronlim02 committed:
Give users a way to find a task

aaronlim02 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/notchatgpt/Ui.java

vikeedough commented on src/main/java/notchatgpt/NotChatGPT.java:
```diff
@@ -0,0 +1,144 @@
+        } else {
+            ui.showTaskList("Tasks found: ", matchingTasks);
+        }
+    }
```
I really like how you wrote the code for handling of the different commands!

vikeedough commented on src/main/java/notchatgpt/Storage.java:
```diff
@@ -0,0 +1,67 @@
+
+package notchatgpt;
+
+import java.io.*;
```
Maybe try not to use wildcard imports?

CoreySiah replied:
In agreement with vikeedough on specifying the exact imports 😁

vikeedough commented on src/main/java/notchatgpt/Storage.java:
```diff
@@ -0,0 +1,67 @@
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(";");
+                switch (parts[0]) {
+                    case "T":
+                        tasks.add(new ToDo(parts[1]));
+                        break;
+                    case "D":
+                        LocalDate by = LocalDate.parse(parts[2]);
+                        tasks.add(new Deadline(parts[1], by));
+                        break;
+                    case "E":
+                        LocalDate from = LocalDate.parse(parts[2]);
+                        LocalDate to = LocalDate.parse(parts[3]);
+                        tasks.add(new Event(parts[1], from, to));
+                        break;
+                }
```
Perhaps add a default case here to catch cases that cannot be recognized if an unrecognized task somehow falls through?

vikeedough reviewed (commented):
Really clean and succinct code! Good job!

reuben-thomas commented on src/main/java/notchatgpt/Deadline.java:
```diff
@@ -0,0 +1,26 @@
+
+public class Deadline extends Task {
+
+  protected LocalDate by;
```
The indentation should be [4 spaces](https://se-education.org/guides/conventions/java/intermediate.html#layout), I think you may have mistakenly set your IDE to 2 instead.

reuben-thomas commented on src/test/java/notchatgpt/ParserTest.java:
```diff
@@ -0,0 +1,42 @@
+package notchatgpt;
+
+import org.junit.jupiter.api.Test;
```
The import statements should be ordered as per the [convention](https://se-education.org/guides/conventions/java/intermediate.html#package-and-import-statements) they provided. You can get IntelliJ to do this automatically for you if you follow [this](https://se-education.org/guides/tutorials/intellijCodeStyle.html).

reuben-thomas commented on src/test/java/notchatgpt/ParserTest.java:
```diff
@@ -0,0 +1,42 @@
+    public void parseTodoDescriptionTest1() {
+        assertEquals("test", parseTodoDescription("todo test"));
+    }
+    @Test
```
It may be a good idea to add spaces in between functions for readability.

reuben-thomas commented on src/main/java/notchatgpt/Task.java:
```diff
@@ -0,0 +1,36 @@
+ */
+package notchatgpt;
+
+public class Task {
```
This is optional, and I'm not sure if it is necessary myself, but would you like to declare this class as abstract so that it can't be instantiated? This task type isn't supported by your agent.

reuben-thomas commented on src/main/java/notchatgpt/Storage.java:
```diff
@@ -0,0 +1,67 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
```
Since storage is specific for storing a TaskList, you may want to give it a more descriptive name, such as `TaskListStorage` or to contain the storage, task and tasklist functionality into a separate `tasklist` package.

reuben-thomas reviewed (commented):
In my opinion, this was a very efficiently written, concise codebase. The only mildly serious fix is the indentation which should be quick. 

I had to really squint to find a possible area to improve in the Code Quality area apart from `Storage.java`.

CoreySiah commented on src/main/java/notchatgpt/NotChatGPT.java:
```diff
@@ -0,0 +1,144 @@
+    }
+
+    private void handleTodo(String input) {
+        if (input.length() <= 5) {
```
Great use of input checks to prevent run-time errors! 😁

CoreySiah commented on src/main/java/notchatgpt/NotChatGPT.java:
```diff
@@ -0,0 +1,144 @@
+        }
+    }
+
+    private void handleFind(String input) {
```
Is it possible to move handleFind() to an existing class? 😁

CoreySiah commented on src/main/java/notchatgpt/Task.java:
```diff
@@ -0,0 +1,36 @@
+    return description;
+  }
+
+  public void setStatusIcon(char c) {
```
Possibly look to have the parameter be a boolean instead of char? 😁

CoreySiah commented on src/main/java/notchatgpt/Ui.java:
```diff
@@ -0,0 +1,42 @@
+
+import java.util.Scanner;
+
+public class Ui {
```
Well written Ui class! 😁

CoreySiah reviewed (commented):
LGTM! Great work; just general comments! 😁

aaronlim02 committed:
Use CheckStyle

aaronlim02 committed:
Merge branch 'branch-A-CheckStyle'

aaronlim02 committed:
Complete JavaFX part 1

aaronlim02 committed:
Complete JavaFX part 2

aaronlim02 committed:
Complete JavaFX part 4

aaronlim02 committed:
Implement GUI to the chatbot

aaronlim02 committed:
Complete Level 10. GUI

aaronlim02 committed:
Merge branch 'branch-Level-10'

aaronlim02 committed:
Update MANIFEST.MF

aaronlim02 committed:
Use Assertions

Add input validation using assertions in readCommand method

This commit introduces input validation using Java's assert keyword in
the readCommand method to ensure that the provided command is neither
null nor empty.

* Added assertions to enforce input constraints:
* * assert s != null ensures that the input string is not null.
* * assert !s.trim().isEmpty() ensures that the input is not an empty
string after trimming.
* Wrapped the readCommand invocation in a try-catch block to handle
AssertionError.
* If an assertion fails, an error message "Command cannot be empty" is
displayed.

aaronlim02 committed:
Improve Code Quality

aaronlim02 committed:
Merge pull request #2 from aaronlim02/branch-A-Assertions

Use Assertions

aaronlim02 committed:
Merge pull request #3 from aaronlim02/branch-A-CodeQuality

Improve Code Quality

aaronlim02 committed:
Merge branch 'master' of https://github.com/aaronlim02/ip

aaronlim02 committed:
Add Easily edit items functionality

aaronlim02 committed:
fix bugs

aaronlim02 committed:
Change Title

aaronlim02 committed:
Change title

aaronlim02 committed:
Merge branch 'master' into C-Update

aaronlim02 committed:
Add command usage guide when command is called

aaronlim02 committed:
Add help command

aaronlim02 committed:
Update README.md

aaronlim02 committed:
Update README.md

aaronlim02 committed:
Update README.md

aaronlim02 committed:
Update README.md

aaronlim02 committed:
Add more javadoc comments
